versioned_docs/version-2020fa/finalproject.md

```diff
@@ -0,0 +1,141 @@
+
+### Milestone 0
+
+**Due: November 10 by 3:59 PM (no slip days)**
+
+Upload to CMS a mock and description of your web app! Your mock can be a picture (handdrawn or typed) and the description should be in `pdf` form. Your mock should include sketches of the different pages or views in your application. Your description should be at least 200 words and elaborate on what your project will be about. Also consider how you will structure your database schema! (What tables will you need, how will they be related, and what will they store). Please zip both of these together to submit to CMS.
+
+### Milestone 1
+
+**Due: December 1 by 3:59 PM**
+
+Get started implementing either the frontend or backend (you choose).
+If you're doing frontend, you can use hardcoded data for now just to get the layout done.
+
+### Milestone 2
+
+**Due: December 8 by 3:59PM**
+
+Work on the other side (if you did frontend for Milestone 1, do backend now and vice versa).
+
+Also have an authentication system set up using Firebase.
+
+### Milestone 3
+
+**Due: December 15 by 3:59PM**
+
+Bridge the frontend and backend together using data fetching. This will involve some full stack work! Also, the app should be fully deployed on Firebase and styled using any CSS you deem necessary. This is your final submission!
+
+:::tip
+Need help? Come to [office hours](/docs/introduction#when-are-office-hours) and we can help you debug. Also feel free to post questions on Piazza!
+:::
+
+## Optional Extensions
+
+If you're looking for an added challenge, consider doing one or more of these extensions! There is no extra credit for these extensions, but it will really enrich your understanding of webdev (and make us very impressed!) :)
+
+1. Use Material-UI or Ionic or Bootstrap (libraries to help with styling!)
+2. Mobile-friendly/Responsive
+3. Use an external API with your backend (note: this doesn't replace your Firebase backend)
+
+## Grading
+
+- Frontend (25%)
+
+  - Use of ES6 & TypeScript
+  - Component structure
+  - Use of hooks
+  - Functional programming
+
+- Backend (25%)
+
+  - Use of ES6 & TypeScript
+  - Has both GET and POST requests
+  - Is connected to a database
+  - Functional programming
+
+- Database/Auth (15%)
+
+  - Data fetching
+  - Data insertion
+  - Authentication (using Firebase auth)
+
+- Styling (5%)
+
+- Deployment (5%)
+
+- Code style and file structure (5%)
+
+- Effort (20%)
+  - This may seem like an arbitrary measure, but we expect most people to
+    get most of the points here; the only way to lose points is if it is
+    clear that not a lot of effort has been put in or you are consistently
+    missing milestones.
+
+## Submission
+
+Submission for each milestone will be a zip file of your code, along with a
+README containing a description of the project, a list of
+group members and netIDs, a link to the deployed site,
+and anything else that you think is important for us
+to know.
+
+As always, do not include your `node_modules`.
```

Comment:
maybe we can also add some tips, or point to [How to lose in CS 2112](https://www.cs.cornell.edu/courses/cs2112/2020fa/handouts/how-to-lose.html)